### PR TITLE
forbid unnamed context

### LINKED ISSUE #56747: kubectl config view --minify=true fails for kubeconfig without current context
Separate cluster kubeconfig files can have unnamed contexts, e.g.

```
apiVersion: v1
kind: Config
clusters:
- name: cluster-8f18
  cluster:
    server: https://foo.bar
users:
- name: admin
  user:
contexts:
- context:
    cluster: cluster-8f18
    user: admin
```

Then `kubectl config view --minify=true` complains that no context is set.

Meanwhile:
```
$ kubectl config get-contexts
CURRENT   NAME      CLUSTER        AUTHINFO   NAMESPACE
*                   cluster-8f18   admin
```

and

```
$ kubectl config current-context
error: current-context is not set
```

and

```
$ kubectl config use-context ""
error: you must specify a current-context
```

All not very consistent.

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:
forbid unnamed contexts with validation

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #56747

**Special notes for your reviewer**:
/assign @sttts @fabianofranz 

**Release note**:

```release-note
Forbid unnamed contexts in kubeconfigs.
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/tools/clientcmd/api/v1/conversion.go [hunk 1]
```diff
@@ -167,6 +168,9 @@ func addConversionFuncs(scheme *runtime.Scheme) error {
 				if err := s.Convert(&curr.Context, newContext, 0); err != nil {
 					return err
 				}
+				if len(curr.Name) == 0 {
+					return fmt.Errorf("you must specify a non-empty context name for %#v", curr)
```
Comment (sttts) [inline staging/src/k8s.io/client-go/tools/clientcmd/api/v1/conversion.go]: non-empty context names are not allowed. "you" sounds strange here.

Comment (dixudx) [reply staging/src/k8s.io/client-go/tools/clientcmd/api/v1/conversion.go]: Make sense. Updated. PTAL.

File: pkg/kubectl/cmd/config/use_context.go [hunk 2]
```diff
@@ -88,7 +88,7 @@ func (o *useContextOptions) complete(cmd *cobra.Command) error {
 
 func (o useContextOptions) validate(config *clientcmdapi.Config) error {
 	if len(o.contextName) == 0 {
-		return errors.New("you must specify a current-context")
+		return errors.New("non-empty context names are not allowed")
```
Comment (liggitt) [inline pkg/kubectl/cmd/config/use_context.go]: this message text is very confusing

File: staging/src/k8s.io/client-go/tools/clientcmd/api/v1/conversion.go [hunk 3]
```diff
@@ -167,6 +168,9 @@ func addConversionFuncs(scheme *runtime.Scheme) error {
 				if err := s.Convert(&curr.Context, newContext, 0); err != nil {
 					return err
 				}
+				if len(curr.Name) == 0 {
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/tools/clientcmd/api/v1/conversion.go]: conversion isn't the right place to do validation like this... validate at point of use

File: staging/src/k8s.io/client-go/tools/clientcmd/api/v1/conversion.go [hunk 4]
```diff
@@ -167,6 +168,9 @@ func addConversionFuncs(scheme *runtime.Scheme) error {
...
+					return fmt.Errorf("non-empty context name for %#v is not allowed", curr)
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/tools/clientcmd/api/v1/conversion.go]: I think this should be "empty context name is not allowed"

Comment (dixudx) [reply staging/src/k8s.io/client-go/tools/clientcmd/api/v1/conversion.go]: my bad.

File: staging/src/k8s.io/client-go/tools/clientcmd/api/v1/conversion.go (hunk 3 above)
Comment (dixudx) [reply staging/src/k8s.io/client-go/tools/clientcmd/api/v1/conversion.go]: For such `conversionFuncs` registered/initialized at `init`, it is quite hard to do the validate at point of use. Will looks quite ugly if added IMO.

Comment (liggitt) [reply staging/src/k8s.io/client-go/tools/clientcmd/api/v1/conversion.go]: Errors in conversion do not surface well. This data should not prevent conversion.

Comment (liggitt) [reply staging/src/k8s.io/client-go/tools/clientcmd/api/v1/conversion.go]: Validate here: https://github.com/dixudx/kubernetes/blob/20da5a0b5f46f10054ea4d3eacaaf760a890cc1b/staging/src/k8s.io/client-go/tools/clientcmd/validation.go

**Comment by dixudx** [comment]
ping @liggitt for another look. Thanks.

**Comment by sttts** [comment]
Updated the release note. Ptal.

**Pull request merged**